hudi-common/src/main/java/org/apache/hudi/common/model/PartialUpdateAvroPayload.java

```diff
@@ -89,6 +89,13 @@
  */
 public class PartialUpdateAvroPayload extends OverwriteNonDefaultsWithLatestAvroPayload {
 
+  /*
+   flag for deleted record combine logic
+   1 preCombine: if delete record is newer, return merged record with _hoodie_is_deleted = true
+   2 combineAndGetUpdateValue:  return empty since we don't need to store deleted data to storage
+  */
+  private boolean isPreCombining = false;
```

Comment:
Using member variable for passing state for direct function calls needs to be avoided. Instead pass the isPreCombining as argument to mergeOldRecord calls.